pallets/pallet-bonded-coins/src/tests/transactions/refund_account.rs

```diff
@@ -0,0 +1,368 @@
+				collateral_id,
+				pool_id.clone(),
+				true,
+				1000
+			));
+			Pools::<Test>::mutate(&pool_id, |details| {
+				details.as_mut().unwrap().collateral_id = collateral_id
+			});
+			// put less than 2*min balance in pool account
+			let total_collateral = 1500;
+			assert_ok!(<Test as crate::Config>::CollateralCurrencies::mint_into(
+				collateral_id,
+				&pool_id.clone(),
+				total_collateral
+			));
+
+			let origin = RawOrigin::Signed(ACCOUNT_01).into();
+
+			assert_ok!(BondingPallet::refund_account(origin, pool_id.clone(), ACCOUNT_01, 0, 1));
+
+			assert_eq!(
+				<Test as crate::Config>::Fungibles::total_balance(DEFAULT_BONDED_CURRENCY_ID, &ACCOUNT_01),
+				0
+			);
+			// each would get half, which is below minimum - should not get transferred
+			assert_eq!(
+				<Test as crate::Config>::Fungibles::total_balance(collateral_id, &pool_id),
+				total_collateral
+			);
+		});
+}
+
+#[test]
+fn refund_account_fails_when_pool_not_refunding() {
+	let pool_details = generate_pool_details(
+		vec![DEFAULT_BONDED_CURRENCY_ID],
+		get_linear_bonding_curve(),
+		false,
+		Some(PoolStatus::Active),
+		Some(ACCOUNT_00),
+		None,
+		None,
+	);
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_01, ONE_HUNDRED_KILT)])
+		.with_pools(vec![(pool_id.clone(), pool_details.clone())])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![
+			(DEFAULT_COLLATERAL_CURRENCY_ID, pool_id.clone(), ONE_HUNDRED_KILT),
+			(DEFAULT_BONDED_CURRENCY_ID, ACCOUNT_01, ONE_HUNDRED_KILT),
+		])
+		.build()
+		.execute_with(|| {
+			let origin: OriginFor<Test> = RawOrigin::Signed(ACCOUNT_01).into();
+
+			// Ensure the refund_account call fails due to pool not being in refunding state
+			assert_err!(
+				BondingPallet::refund_account(origin, pool_id.clone(), ACCOUNT_01, 0, 1),
+				Error::<Test>::NotRefunding
+			);
+		});
+}
+
+#[test]
+fn refund_account_no_balance() {
+	let pool_details = generate_pool_details(
+		vec![DEFAULT_BONDED_CURRENCY_ID],
+		get_linear_bonding_curve(),
+		true,
+		Some(PoolStatus::Refunding),
+		Some(ACCOUNT_00),
+		Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+		Some(ACCOUNT_00),
+	);
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	ExtBuilder::default()
+		.with_pools(vec![(pool_id.clone(), pool_details.clone())])
+		.with_native_balances(vec![(ACCOUNT_00, ONE_HUNDRED_KILT), (ACCOUNT_01, ONE_HUNDRED_KILT)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![
+			(DEFAULT_COLLATERAL_CURRENCY_ID, pool_id.clone(), ONE_HUNDRED_KILT),
+			(DEFAULT_BONDED_CURRENCY_ID, ACCOUNT_00, ONE_HUNDRED_KILT),
+		])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_01).into();
+
+			// Ensure the refund_account call fails when there is no balance to be
+			// refunded
+			assert_storage_noop!(assert!(BondingPallet::refund_account(
```

Comment:
The inner assert! seems a bit obsolete.

Reply:
This is not addressed yet.

Reply:
We must handle the result in some form of course.

Reply:
Or I can just assert a FundsUnavailable error ?

Reply:
84852bef459a59d75a8b75e4d55888f06cb2f3fc